apis/workload/v1alpha1/types.go

```diff
@@ -198,8 +198,8 @@ type KubernetesApplicationResourceStatus struct {
 	// State of the application.
 	State KubernetesApplicationResourceState `json:"state,omitempty"`
 
-	// Cluster to which this application has been scheduled.
-	Cluster *KubernetesClusterReference `json:"clusterRef,omitempty"`
+	// Target to which this application has been scheduled.
+	Target *KubernetesTargetReference `json:"targetRef,omitempty"`
```

Comment:
Not related to this PR but I think this should be moved to spec as well at some point. In general, status should be reproducible from (spec + observation). https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

Reply:
I don't think this is as black and white as it may seem; I believe the original idea behind putting this in the status was that it was only expected to be set by the scheduler (based on the clusterselector). In this case the status _is_ reproducable - if it were lost the workload would be rescheduled (not necessarily to the same cluster, but still to an an appropriate cluster).

That said:

* I think most of our more modern selector-and-ref patterns put both in the spec.
* If we wanted to support a user explicitly picking a cluster spec would be better for the ref.

Reply:
@negz In your second bullet do you mean explicitly picking a target? Or do you intend that we update this to be able to reference a cluster directly as well?